### DISCUSSION TITLE
Badges for Table of Contents and h tags

### DISCUSSION THREAD

**Discussion by onurusluca** [withastro/starlight#1024] [Q&A] [answered]
### What version of `starlight` are you using?

3.4.0

### What is your idea?

It would be great to have badges, like on the sidebar, for table of contents and the h tag for that item.

Example:
![image](https://github.com/withastro/starlight/assets/51825926/6dbeca78-4d93-4ad4-ad6a-00c69ac46ea5)
![image](https://github.com/withastro/starlight/assets/51825926/0e1d99a1-20cc-444d-8858-fa2b841cb096)
![image](https://github.com/withastro/starlight/assets/51825926/19d9bf4e-36a3-4fc9-8510-012e9de80e8b)
![image](https://github.com/withastro/starlight/assets/51825926/93cdb173-1193-4e51-8fed-e3bf6a21bb8b)

Related:
https://starlight.astro.build/guides/sidebar/#badges

We can [override ](https://starlight.astro.build/guides/customization/#table-of-contents) the table of contents and add some stuff ourselves but it's too much work to write manually.

If we can decide on a direction, I can implement it. I wanted to ask the maintainers first.

### Why is this feature necessary?

To show badges for the content inside a page other than the currently available sidebar badges.

Sidebar badge:
![image](https://github.com/withastro/starlight/assets/51825926/37394c9b-0b37-41ef-b085-429088af4f29)

### Participation

- [X] I am willing to submit a pull request for this feature.

**Comment by delucis** [comment]
Thanks for the idea @onurusluca! This would indeed be cool, but we might be blocked by how the table of contents is generated. We use Astro’s [`headings` array](https://docs.astro.build/en/reference/api-reference/#render) generated when rendering the document. This gives plain text content and importantly does not include components from MDX (see https://github.com/withastro/starlight/issues/561 and https://github.com/withastro/astro/issues/8240). So while we could expose a `<Badge>` component for use in MDX pages, this would still not make it to the table of contents.

**Comment by HiDeoo** [comment, marked as answer]
A similar feature is available in the [`starlight-heading-badges`](https://github.com/HiDeoo/starlight-heading-badges) Starlight plugin which relies on the [generic directives proposal](https://talk.commonmark.org/t/generic-directives-plugins-syntax/444) syntax.

<img width="980" alt="image" src="https://github.com/user-attachments/assets/b926016f-7778-456c-9143-5b014a365d88">

**Comment by onurusluca** [comment]
Thank you! Closing.
